src/rdf4cpp/datatypes/registry/DatatypeConversion.hpp

```diff
@@ -340,38 +340,38 @@ consteval ConversionTable auto make_conversion_table() {
         using source_type = Type;
         using target_type = Type;
 
-        inline static typename target_type::cpp_type convert(typename source_type::cpp_type const &value) noexcept {
+        static typename target_type::cpp_type convert(typename source_type::cpp_type const &value) noexcept {
             return value;
         }
 
-        inline static nonstd::expected<typename source_type::cpp_type, DynamicError> inverse_convert(typename target_type::cpp_type const &value) noexcept {
+        static nonstd::expected<typename source_type::cpp_type, DynamicError> inverse_convert(typename target_type::cpp_type const &value) noexcept {
             return value;
         }
     };
 
-    using level_0_table = typename decltype(make_promotion_layer<Type>())::template prepend<IdConversion>;
+    using Level0Table = util::concat_t<util::type_list<IdConversion>, decltype(make_promotion_layer<Type>())>;
 
     if constexpr (!SubtypedLiteralDatatype<Type>) {
-        return mz::type_list<level_0_table>{};
+        return util::type_list<Level0Table>{};
     } else {
         using s_table = decltype(make_subtype_layer<Type>());
 
         // generate the linear promotion table for each supertype
-        auto const other_level_tables = util::type_list_map<s_table>([]<ConversionEntry ToSuper>() noexcept {
-            using level_p_table = decltype(make_promotion_layer<typename ToSuper::target_type>());
+        using OtherLevelTables = util::transform_t<s_table, []<ConversionEntry ToSuper>(std::type_identity<ToSuper>) {
```

Comment:
why do we need to add `std::type_identity<ToSuper>` as an argument?

Reply:
yes